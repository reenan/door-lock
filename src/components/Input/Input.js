import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './Input.scss'

class Input extends Component {
  render() {
    const { type, label, required, value, className, maxLength, onChange } = this.props

    return (
      <div className={`input ${className}`}>
        <input type={type} required={required} defaultValue={value}
          maxLength={maxLength} onChange={onChange} />

        <label>{label}</label>
      </div>
    )
  }
}

Input.propTypes = {
  className: PropTypes.string,
  type: PropTypes.string,
  required: PropTypes.bool,
  value: PropTypes.string,
  label: PropTypes.string,
  maxLength: PropTypes.number,
  onChange: PropTypes.func.isRequired,
}

Input.defaultProps = {
  className: '',
  type: 'text',
  required: false,
  value: null,
  label: null,
  maxLength: null,
}

export default Input
