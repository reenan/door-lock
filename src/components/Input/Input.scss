@import '../../resources/_mixins.scss';

div.input {
  position: relative;

  input {
    @include transition(border 400ms);
    @include box-sizing(border-box);

    display: block;
    outline: 0;
    width: 100%;
    border: none;
    font-size: 18px;
    height: 30px;
    color: black;
    border-bottom: 1px solid rgba(black, 0.2);
    background: transparent;
    padding: 0 5px;

    &:focus {
      border-bottom: 1px solid rgba(black, 0.5);
    }

    &:valid, &:focus, &:-webkit-autofill {
      & + label {
        font-size: 12px;
        color: #25cb68;
        top: -13px;
      }
    }
  }

  label {
    @include transition(all 400ms);
    pointer-events: none;
    color: gray;
    font-size: 14px;
    position: absolute;
    top: 0;
    left: 5px;
  }
}