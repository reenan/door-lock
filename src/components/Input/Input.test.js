import React from 'react'
import Input from './Input'

import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const component = renderer.create(
    <Input onChange={() => {}} />
  )

  expect(component).toMatchSnapshot()
})

it('renders without crashing with value', () => {
  const component = renderer.create(
    <Input value='Mock' onChange={() => {}} />
  )

  expect(component).toMatchSnapshot()
})
