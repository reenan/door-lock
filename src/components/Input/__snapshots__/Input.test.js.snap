// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="input "
>
  <input
    defaultValue={null}
    maxLength={null}
    onChange={[Function]}
    required={false}
    type="text"
  />
  <label />
</div>
`;

exports[`renders without crashing with value 1`] = `
<div
  className="input "
>
  <input
    defaultValue="Mock"
    maxLength={null}
    onChange={[Function]}
    required={false}
    type="text"
  />
  <label />
</div>
`;
